src/screen.c

```diff
@@ -5496,15 +5496,6 @@ win_line(
 	if (c == NUL)
 	{
 #ifdef FEAT_SYN_HL
-	    if (eol_hl_off > 0 && vcol - eol_hl_off == (long)wp->w_virtcol
```

Comment:
I must admit that I do not understand the intended purpose of this code block. This was added in vim/vim@a443af8 (http://ftp.vim.org/pub/vim/patches/7.1/7.1.152). Unfortunately, neither of those provide very much context for the change.

I was unable to find any problematic interactions between hls and cursorcolumn (mentioned in the patch diff) that occur when this block is removed.